### PR TITLE
Release 1.7

### LINKED ISSUE #14112: Release 1.7
### Prepare the project

Do some housekeeping on GitHub in the [main repo](https://github.com/OrchardCMS/OrchardCore).

- [x] Close remaining issues for the version (including merging corresponding pull requests if suitable) or assign them to the next one.
- [ ] Assign all issues that were closed for an upcoming version (including a wildcard version like "1.0.x") to this version (milestone).

### Prepare the code

Update the source so everything looks like on the new version.

- [x] Create a `release/<version name>` branch out of `main`, e.g. `release/1.8`.
- [ ] Check the [release_ci](https://github.com/OrchardCMS/OrchardCore/blob/main/.github/workflows/release_ci.yml) workflow is using the expected .NET version to build the Docker images.
- [x] Create a `release/<version name>` branch out of `main`, e.g. `release/1.3.1`.
- [x] Update the `OrchardCore.Commons.props` file with `<VersionSuffix></VersionSuffix>` such that preview build numbers are not injected in packages. Verify the `VersionPrefix` tag matches the released version.
- [x] Update module versions in `src/OrchardCore/OrchardCore.Abstractions/Modules/Manifest/ManifestConstants.cs`.
- [x] Update the version in the command lines in from all documentation files.
- [x] Create a new milestone.

### Test the release

Make sure everything works all right.

- [x] Make sure that [`OrchardCore.Samples` works](https://github.com/OrchardCMS/OrchardCore.Samples).
- [x] Test the [guides](https://docs.orchardcore.net/en/latest/docs/guides/) with the NuGet packages from the Cloudsmith feed (branches under `release/` are automatically published too). Test at least the following guides:
    - [Creating a modular ASP.NET Core application](https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/)
    - [Creating an Orchard Core CMS website](https://docs.orchardcore.net/en/latest/docs/guides/create-cms-application/)
    - [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/)
- [x] If there's a more recent major version of Red Hat Enterprise Linux (like v10 after v9) that Orchard Core was certified for, then re-certify it. See Orchard's [Red Hat Ecosystem Catalog profile](https://catalog.redhat.com/software/applications/detail/223797) for the version it was certified for, the [Red hat Customer Portal](https://access.redhat.com/articles/3078) for the latest released version, and [https://docs.orchardcore.net/en/latest/docs/topics/red-hat-ecosystem-catalog-certification/](our certification guide) for the steps to renew the certification.

### Prepare and publish Orchard Core Translations

Update everything in the [Translations project](https://github.com/OrchardCMS/OrchardCore.Translations). Only do this once all the code changes are done since localized strings can change until then.

- [x] Update .po files with [PoExtractor](https://github.com/lukaskabrt/PoExtractor). This will also update [Crowdin](https://crowdin.com/project/orchard-core).
- [x] Publish the new version on NuGet.
- [x] Update the `OrchardCore.Translations.All` package reference in the main repo's _src/OrchardCore.Build/Dependencies.props_ file to refer to the new NuGet package.

### Prepare the documentation

Update the docs so they contain information about the new release so once the release is out you'll just need to point to new information.

- [ ] Create a new __Draft__ Release with a tag `vx.y.z` that is created when the release is published. Auto-generate release notes.
- [ ] Create release notes in a specific documentation section. You can take the previous release notes as a template.
    - Overview of the release's highlights and goals. What do you want people to remember this release for?
    - Prerequisites. What framework version do you need, anything else to work with Orchard?
    - Upgrade steps, any migration necessary from previous versions, breaking changes.

### Publish the release

Do the harder parts of making the release public. This should come after everything above is done.

- [ ] Merge `release/<version name>` to `main`.
    - Merges to `main` need two approvals so you'll need to create a pull request.
    - Merge it as a merge commit, not squash merge.
- [ ] Publish the Draft release.
- [ ] Test the [guides](https://docs.orchardcore.net/en/latest/docs/guides/) with the packages now automatically published to NuGet. Test at least the following guides:
    - [Creating a modular ASP.NET Core application](https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/)
    - [Creating an Orchard Core CMS website](https://docs.orchardcore.net/en/latest/docs/guides/create-cms-application/)
    - [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/)
- [ ] Update [Try Orchard Core](https://github.com/OrchardCMS/TryOrchardCore).

### Publicize the release

Let the whole world know about our shiny new release. Savor this part! These steps will make the release public so only do them once everything else is ready.

- [ ] Update the documentation to mention the version in all places where the latest version is referenced, for example, but not limited to (do a search for the package version string): [Status in the root README](https://docs.orchardcore.net/en/latest/#status), CLI templates, commands, the [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/) guide.
- [ ] Update the tagged [release](https://github.com/OrchardCMS/OrchardCore/releases) on GitHub: Change its title to something more descriptive (e.g. "Orchard Core 1.0.0 RC 2"), add a link in its description to the release notes in the documentation (something like `For details on this version see the [release notes in the documentation](link here).`). Add a link to this release under [Status in the root README](https://docs.orchardcore.net/en/latest/#status).
- [ ] Publish a blog post on the website.
- [ ] Ask to publish a blog post on [DevBlogs](https://devblogs.microsoft.com/).
- [ ] Ask to publish a blog post on [.NET Foundation News](https://dotnetfoundation.org/news).
- [ ] Tweet

### After the release is done

- [x] Create a new milestone with the next release number.
- [ ] Create a new release notes documentation file for the next version in the OrchardCore.Docs project. (e.g, `/releases/1.8.0.md`).
- [ ] Update the `OrchardCore.Commons.props` file with the next release number, and `<VersionSuffix>preview</VersionSuffix>` such that preview builds use the new one.

### PR DESCRIPTION
**Pull request by sebastienros**
Fix #14112

### CODE REVIEW

**Comment by hyzx86** [comment]
Propose to merge this PR 
https://github.com/OrchardCMS/OrchardCore/pull/14209

**Comment by hishamco** [comment]
@hyzx86 it's already merged :)

**Comment by agriffard** [comment]
@MikeAlhayek https://github.com/OrchardCMS/OrchardCore.Translations/pull/101 ready to be merged with latest translations.

**Comment by MikeAlhayek** [comment]
@agriffard I guess I do not have permission on the OrchardCore.Translations. I can't create a release. Can you please create 1.7.1 release or give me access to do it.

**Comment by agriffard** [comment]
> @agriffard I guess I do not have permission on the OrchardCore.Translations. I can't create a release. Can you please create 1.7.1 release or give me access to do it.

PR Merged (will be deployed on preview). 
Permission added on the repo for your user.
I let you create the 1.7.1 Release.

**Comment by MikeAlhayek** [comment]
@agriffard thank you. I updated the translation. Then need more time to be indexed. Now, we need need to rebase this branch then merge it into main and I'll create the 1.7 OC release. If you, @sebastienros or @jtkech wants to do the rebasing/merging soon I can release 1.7 soon.

**Comment by hishamco** [comment]
@MikeAlhayek  why there's `1.7.1` for OC.Translations?

**Comment by agriffard** [comment]
> @MikeAlhayek why there's `1.7.1` for OC.Translations?

Because other PRs with strings needing translations from Crowdin have been merged after the 1.7.0 of OC.Translations.

**Comment by sebastienros** [comment]
> Because other PRs with strings needing translations from Crowdin have been merged after the 1.7.0 of OC.Translations.

Actually, not really. Since there are no custom translations for these new strings there is absolutely no difference in OC to use 1.7.0 or 1.7.1 versions. If even 1 day after we ship there are new actual translations we can always ship a new 1.7.2 with these.

If OC had been shipped without the new entries, the way it works is that the string in the view is used, the English one, which is exactly the same as with 1.7.1 but the English entry that exists.

**Comment by MikeAlhayek** [comment]
@sebastienros I just noticed you did a force-push few hours ago. Can you please do another rebase? Once done, let me know if I can release 1.7

**Comment by sebastienros** [comment]
We don't really need to rebase or merge in this case: we are not running functional tests again, and there are no functional changes in this branch, tests could be done in main too at that point. When we are ready we can just merge this PR in main. 

But let's wait for Tuesday since we got the reflex to not ship just before a WE in case there is an issue with the release or a huge bug to fix.

File: src/docs/topics/publishing-releases/README.md [hunk 1]
```diff
@@ -84,9 +84,4 @@ Let the whole world know about our shiny new release. Savor this part! These ste
 - [ ] Ask to publish a blog post on [.NET Foundation News](https://dotnetfoundation.org/news).
 - [ ] Tweet
 
-### After the release is done
```
Comment (sebastienros) [inline src/docs/topics/publishing-releases/README.md:87]: Why was it removed?

Comment (MikeAlhayek) [reply src/docs/topics/publishing-releases/README.md:87]: both steps already exists in the same document is earlier steps

**Comment by Skrypt** [comment]
Azure.Storage.Blobs version should be 12.17.0 instead of 12.16.0 in the release notes.

**Comment by agriffard** [comment]
> Azure.Storage.Blobs version should be 12.17.0 instead of 12.16.0 in the release notes.

There has been a second PR for the 12.17.0 version.

What you call release notes is a description auto-generated when the Release is created with the list of PRs merged since the previous release.

**Pull request merged**